lib/comments.ts

```diff
@@ -1,19 +1,16 @@
-var assert = require("assert");
-var types = require("./types");
+import assert from "assert";
+import types from "./types";
 var n = types.namedTypes;
 var isArray = types.builtInTypes.array;
 var isObject = types.builtInTypes.object;
-var linesModule = require("./lines");
-var fromString = linesModule.fromString;
-var Lines = linesModule.Lines;
-var concat = linesModule.concat;
-var util = require("./util");
-var comparePos = util.comparePos;
-var childNodesCacheKey = require("private").makeUniqueKey();
+import { Lines, concat } from "./lines";
+import { comparePos, fixFaultyLocations } from "./util";
+import { makeUniqueKey } from "private";
+var childNodesCacheKey = makeUniqueKey();
 
 // TODO Move a non-caching implementation of this function into ast-types,
 // and implement a caching wrapper function here.
-function getSortedChildNodes(node, lines, resultArray) {
+function getSortedChildNodes(node: any, lines: any, resultArray?: any) {
```

Comment:
Is there some automated transform that helps with this, or was this all by hand? 😬

Reply:
Did these ones by hand, but I'm working on a tool that automates patterns like this. It's called ts-migrate (super alpha at this point) and its a collection of codemods and scripts for migrating codebases to TS. Here's the code that would have added these https://github.com/brieb/ts-migrate/blob/dc324e1aefa6952c78b3bd092ca4b34ff8eb64c4/src/plugins/explicit-any/withExplicitAny.ts

Reply:
Cool!